src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -418,11 +429,15 @@ OperationalQueryAllocator::Allocator * AdvertiserMinMdns::FindEmptyOperationalAl
 
 CHIP_ERROR AdvertiserMinMdns::Advertise(const OperationalAdvertisingParameters & params)
 {
+
     char nameBuffer[Operational::kInstanceNameMaxLength + 1] = "";
 
     /// need to set server name
     ReturnErrorOnFailure(MakeInstanceName(nameBuffer, sizeof(nameBuffer), params.GetPeerId()));
 
+    // about to clear all records below, clear them
```

Comment:
I don't think we clear all records...  We're only clearing operational records, right?  And only for the specific operational fabric `params` represents.  Other operational fabrics, or commissionable records are not affected...

I think the code here is correct, in that we are advertising everything we have as TTL=0, then wiping some of them out, replacing them with something else, then advertising everything we have (old and new) with normal TTL.  But the comment could be clearer.